rust/lance/benches/ivf_pq.rs

```diff
@@ -93,6 +93,7 @@ fn bench_ivf_pq_index(c: &mut Criterion) {
                     VectorIndexParams::ivf_pq(ivf_partition, 8, pq, false, MetricType::L2, 50);
 
                 dataset
+                    .clone()
```

Comment:
I think my brain is just not working today.  Is this benchmark going to create many many different versions of the dataset (one per iteration)?

Reply:
Yes, but that's what it was doing before anyways :)

Prior to this, `create_index` returned the new version of the dataset and left the current one alone. This threw away the return value, so every iteration it was back at the old version of the dataset. Adding the clone here gives equivalent behavior.